bindings/compile.ts

```diff
@@ -0,0 +1,194 @@
+import assert from 'assert';
+
+import { isNil } from '../common/helpers';
+import { bindSolcMethod } from './helpers';
+
+export function setup (solJson, core) {
+  return {
+    compileJson: compileJson(solJson),
+    compileJsonCallback: compileJsonCallback(solJson, core),
+    compileJsonMulti: compileJsonMulti(solJson),
+    compileStandard: compileStandard(solJson, core)
+  };
+}
+
+/**********************
+ * COMPILE
+ **********************/
+
+/**
+ * Returns a binding to the solidity compileJSON method.
+ *
+ * @param solJson
+ */
+function compileJson (solJson) {
+  return bindSolcMethod(solJson, 'compileJSON',
+    'string', ['string', 'number'], null);
+}
+
+/**
+ * Returns a binding to the solidity compileJSONMulti method.
+ *
+ * @param solJson
+ */
+function compileJsonMulti (solJson) {
+  return bindSolcMethod(solJson, 'compileJSONMulti',
+    'string', ['string', 'number'], null);
+}
+
+/**
+ * Returns a binding to the solidity compileJSONCallback method.
+ *
+ * @param solJson
+ */
+function compileJsonCallback (solJson, core) {
+  const compileInternal = bindSolcMethod(solJson, 'compileJSONCallback',
+    'string', ['string', 'number', 'number'], null);
+
+  if (isNil(compileInternal)) return compileInternal;
+
+  return function (input, optimize, readCallback) {
+    return runWithCallbacks(solJson, core, readCallback, compileInternal, [input, optimize]);
+  };
+}
+
+/**
+ * Returns a binding to the solidity solidity_compile method with a fallback to
+ * compileStandard.
+ *
+ * @param solJson
+ * @param core
```

Comment:
These should include parameter descriptions.

Reply:
They could but they would have self describing types in the following PR so I did not see it as a requirement. I can add the additional comments if you feel they are required here.

Reply:
I just think that these are pointless if they do not add any extra info on top of name and type. We already have name in the signature and we'll have type as well. If we add them, we should at least try to make them informative.

These two maybe are not best examples because they have unique types (though `SolJson` interface does not have a general description so saying that it's the compiler binary is still useful). But `bindCopyToCString()` for example does not have descriptions for parameters either even though `str` and `ptr` are not very informative on their own.